apps/smartling/lambda/src/index.ts

```diff
@@ -107,10 +116,20 @@ export function makeApp(fetchFn: any, issuer: any) {
     );
   });
 
-  app.use(
-    '/frontend',
-    express.static(path.dirname(require.resolve('@contentful/smartling-frontend')))
-  );
+  if (process.env.LOCAL_DEV === 'true' && process.env.FRONTEND_URL) {
+    // in development mode, proxy requests to the frontend development server running at FRONTEND_URL
+    app.use('/frontend', requestProxier(process.env.FRONTEND_URL));
+
+    // in development mode, static assets are expressed as relative paths, which is served
```

Comment:
If they are relative paths, they should load from `/frontend/static`, right? Shouldn't they only load from `/static` if they are absolute?

Reply:
Good question. I was confused by this at first as well.

To explain by way of example:

In "production" mode the /frontend/index.html returns the following HTML document:

```html
<!doctype html><html><head><meta charset="utf-8"/><script defer="defer" src="./static/js/main.f1d912ca.js"></script><link href="./static/css/main.846e3b9c.css" rel="stylesheet"></head><body><div id="root"></div></body></html>
```

That's from the build (of smartling/frontend). The file paths are relative and thus live under the frontend path. e.g. the first JS file would be served from `/frontend/static/js/main.f1d912ca.js`.

In "development" mode the development server (which we proxy in this PR) the request would give us a different result:

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
  <script defer src="/static/js/bundle.js"></script></head>
  <body>
    <div id="root"></div>
  </body>
</html>
```

Note that the HTML is totally different (not minified) and also only serves up the main static bundle file.

This file lives not at a relative path but rather an absolute path. So that's why we have to set up the secon route in development mode for `/static/*`.

Reply:
What confuses me is that comment

> in development mode, static assets are expressed as relative paths

Shouldn't it be

> in development mode, static assets are expressed as **absolute** paths

An absolute path is `/static`, a relative path is `./static`.

Side note: By default also in prod absolute paths would be used, but we change that with `homepage: '.'` in package.json

Reply:
Ah good catch. The comment itself is wrong / confusing. What I meant to say was "just a path" v.s a full URL.

I.e. they are served as src="/static/somejs.js" instead of src="http://localhost:3030/static/somejs.js". Since the frontend is running on http://localhost:8080 the actual URL request is to http://localhost:8080/static/somejs.js. Hence the needs for a second proxy under the /static root path.